src/tools/pbs_output.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+"""
+/*
+#  Copyright (C) 1994-2017 Altair Engineering, Inc.
+#  For more information, contact Altair at www.altair.com.
+#
+#  This file is part of the PBS Professional ("PBS Pro") software.
+#
+#  Open Source License Information:
+#
+#  PBS Pro is free software. You can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License as published by the Free
+#  Software Foundation, either version 3 of the License, or (at your option) any
+#  later version.
+#
+#  PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+#  PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License along
+#  with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#  Commercial License Information:
+#
+#  The PBS Pro software is licensed under the terms of the GNU Affero General
+#  Public License agreement ("AGPL"), except where a separate commercial license
+#  agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+#
+#  Altair’s dual-license business model allows companies, individuals, and
+#  organizations to create proprietary derivative works of PBS Pro and distribute
+#  them - whether embedded or bundled with other software - under a commercial
+#  license agreement.
+#
+#  Use of Altair’s trademarks, including but not limited to "PBS™",
+#  "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+#  trademark licensing policies.
+ *
+ */
+"""
+__doc__ = """
+This script is not intended to be run by users or Administrators.
+It is run by the Cray RUR system as an "output plugin".
+"""
+
+import os
+import sys
+sys.path.append('/opt/cray/rur/default/bin')
+from rur_plugins import rur_output_args, get_plugin_name, rur_errorlog
+
+
+def outname(jobid):
+    # Create the pathname to write the RUR data for PBS.
+    # By default it will be "/var/spool/PBS/spool/<jobid>.rur"
+    home = "PBS_HOME"
+    dir = "/var/spool/pbs"
```

Comment:
dont hard code the path. If CONF file is not found then exit with error

Reply:
Thats the default value. We first check for PBS_HOME in os environment if not found we read it from conf file.

Reply:
I see that, if PBS_CONF_FILE  is not found then code continues with "/var/spool/pbs". I feel it is wrong.

Reply:
Its more of an initialization. In the end code does "if os.path.isdir(dir)" and on failure to find this directory it raises an exception.